cmd/triggerRun/cmd/root_test.go

```diff
@@ -82,3 +95,154 @@ X-Header: testheader
+	trBytes, err := json.Marshal(taskRunTemplate)
+	if err != nil {
+		t.Fatalf("failed to marshall taskrun to json: %v", err)
+	}
+
+	triggerTemplate := triggersv1.TriggerTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "triggerSpecTemplate",
+			Namespace: "default",
+		},
+		Spec: triggersv1.TriggerTemplateSpec{
+			Params: []triggersv1.ParamSpec{{
+				Name: "foo",
+			}},
+			ResourceTemplates: []triggersv1.TriggerResourceTemplate{{
+				RawExtension: runtime.RawExtension{Raw: trBytes},
+			}},
+		},
+	}
+
+	triggerBinding := v1alpha1.TriggerBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "triggerSpecBinding1",
+			Namespace: "default",
+		},
+		Spec: triggersv1.TriggerBindingSpec{
+			Params: []triggersv1.Param{{
+				Name:  "foo",
+				Value: "bar",
+			}},
+		},
+	}
+
+	wantTaskRun := pipelinev1alpha1.TaskRun{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "tekton.dev/v1alpha1",
+			Kind:       "TaskRun",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-taskrun",
+			Namespace: "default",
+			Labels: map[string]string{
+				"someLabel": "bar", // replaced with the value of foo from bar
+			},
+		},
+		Spec: pipelinev1alpha1.TaskRunSpec{
+			TaskRef: taskRunTemplate.Spec.TaskRef, // non-existent task; just for testing
+		},
+	}
+	wantTrBytes, err := json.Marshal(wantTaskRun)
+	if err != nil {
+		t.Fatalf("failed to marshal wantTaskrun: %v", err)
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    []json.RawMessage
+		wantErr bool
+	}{
+		{
+			name: "testing-name",
+			args: args{
+				t: &v1alpha1.Trigger{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "TriggerRun",
+						APIVersion: "tekton.dev/v1alpha1"},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "my-triggerRun",
+					},
+					Spec: v1alpha1.TriggerSpec{
+						Bindings: []*v1alpha1.TriggerSpecBinding{
+							{Name: "triggerSpecBinding1"}, // These should be references to TriggerBindings defined below
+						},
+						Template: v1alpha1.TriggerSpecTemplate{
+							Name: "triggerSpecTemplate", // This should be a reference to a TriggerTemplate defined below
+						},
+					},
+				},
+
+				request: r,
+				event:   eventBody,
+				resources: test.Resources{
+					// Add any resources that we need to create with a fake client
+					TriggerBindings:  []*v1alpha1.TriggerBinding{&triggerBinding},
+					TriggerTemplates: []*triggersv1.TriggerTemplate{&triggerTemplate},
+				},
+			},
+			want: []json.RawMessage{wantTrBytes},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			eventID := "some-id"
+			logger, _ := zap.NewProduction()
+			eventLog := logger.Sugar()
+			client := getFakeTriggersClient(t, tt.args.resources)
+			got, err := processTriggerSpec(client, tt.args.t, tt.args.request, tt.args.event, eventID, eventLog)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("processTriggerSpec() error = %v. wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
minor: we usually prefer to use cmp.Diff instead of reflect.DeepEqual and print out the diff e.g. https://github.com/tektoncd/triggers/blob/master/pkg/sink/sink_test.go#L141